### PR TITLE
feat: add xgboost handler

### PR DESCRIPTION
**Pull request by machow**
Addresses https://github.com/rstudio/vetiver-python/issues/27, by Implementing xgboost handler.

TODO:

- [ ] : How (or whether) to support XGBClassifier? May already work using sklearn handler
- [ ] : Add tests

### CODE REVIEW

File: vetiver/handlers/xgboost.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    xgboost_exists = False
+
+
+class XGBoostHandler(VetiverHandler):
+
+    base_class = xgboost.Booster
+
+    def describe(self):
+        """Create description for sklearn model"""
+        desc = f"Scikit-learn {self.model.__class__} model"
```
Comment (has2k1) [inline vetiver/handlers/xgboost.py:19]: xgboost!

**Comment by machow** [comment]
@isabelizimm looks like it's passing on CI now. If it's useful, I don't mind doing...

* research into supporting XGBClassifier (and any handler wrangling needed)
* fleshing out tests (should I just make them mirror the ones for sklearn?)

**Comment by machow** [comment]
@isabelizimm, once the changes for https://github.com/rstudio/vetiver-python/issues/67 are in, let me know, and I can rebase / get this code up to date!

**Comment by isabelizimm** [comment]
A lot of the infrastructure around handlers has changed since this PR was created, so I will am closing it in favor of a fresh PR.

**Pull request closed without merging**